core/trino-spi/src/main/java/io/trino/spi/exchange/Exchange.java

```diff
@@ -72,30 +71,12 @@
      * Returns a future containing handles to be used to read data from an exchange.
      * <p>
      * Future must be resolved when the data is available to be read.
-     * <p>
-     * The implementation is expected to return one handle per output partition (see {@link ExchangeSink#add(int, Slice)})
-     * <p>
-     * Partitions can be further split if needed by calling {@link #split(ExchangeSourceHandle, long)}
      *
      * @return Future containing a list of {@link ExchangeSourceHandle} to be sent to a
      * worker that is needed to create an {@link ExchangeSource} using {@link ExchangeManager#createSource(List)}
      */
     CompletableFuture<List<ExchangeSourceHandle>> getSourceHandles();
 
-    /**
-     * Splits an {@link ExchangeSourceHandle} into a number of smaller partitions.
-     * <p>
-     * Exchange implementation is allowed to return {@link ExchangeSourceHandle} even before all the data
-     * is written to an exchange. At the moment when the method is called it may not be possible to
-     * complete the split operation. This methods returns a {@link ExchangeSourceSplitter} object
-     * that allows an iterative splitting while the data is still being written to an exchange.
-     *
-     * @param handle returned by the {@link #getSourceHandles()}
-     * @param targetSizeInBytes desired maximum size of a single partition produced by {@link ExchangeSourceSplitter}
-     * @return {@link ExchangeSourceSplitter} to be used for iterative splitting of a given partition
-     */
-    ExchangeSourceSplitter split(ExchangeSourceHandle handle, long targetSizeInBytes);
```

Comment:
I am not convinced that we should drop this one. It make the SPI contract not obvious. What is the expected size of source handles to be returned by `getSourceHandles` now. What if exchange implementation can split the data into arbitrarly small chunks. Should it return 1M handles each serving 10 bytes of data or 1K each serving 10K?

If we drop split maybe we should pass a hint `targetSourceHandleSizeInBytes` to `getSourceHandles`?

Reply:
Discussed offline